ui/common/src/common.ts

```diff
@@ -129,3 +129,20 @@ export function $as<T>(cashOrHtml: Cash | string): T {
 export function myUserId(): string | undefined {
   return document.body.dataset.user;
 }
+
+export function repeater(f: () => void, e: Event, additionalStopCond?: () => boolean): void {
+  const delay = (function* () {
+    yield 500;
+    for (let d = 350; ; ) yield Math.max(100, (d *= 14 / 15));
+  })();
+  const repeat = () => {
+    f();
+    timeout = setTimeout(repeat, delay.next().value!);
+    if (additionalStopCond?.()) clearTimeout(timeout);
+  };
+  let timeout = setTimeout(repeat, delay.next().value!);
+  f();
+  if (additionalStopCond?.()) clearTimeout(timeout);
+  const eventName = e.type === 'touchstart' ? 'touchend' : 'mouseup';
```

Comment:
I see that `repeat()` does this:
```js
    f();
    timeout = setTimeout(repeat, delay.next().value!);
    if (additionalStopCond?.()) clearTimeout(timeout);
```
and the code from lines 143 to 145 does this:
```js
  let timeout = setTimeout(repeat, delay.next().value!);
  f();
  if (additionalStopCond?.()) clearTimeout(timeout);
```
Which seems to be the exact same thing? Can we replace these 3 lines with a simple call to `repeat()`?

Reply:
@ornicar Updated to do that. The reason I didn't before is that `components.ts` did it in a similar way (screenshot), and it makes a very slight difference for time. On my computer, `f()` usually spends in the neighbourhood of ~8 milliseconds for the analysis board, and somewhere around ~2 ms for the in-progress game board. So setting the timeout after the initial `f()` will cause this amount of additional delay for the first replayed move.

![image](https://github.com/user-attachments/assets/aaca4545-50bf-473d-a618-365a51b92ba1)